apps/sms/views/conversation/js/compose.js

```diff
@@ -51,12 +48,15 @@ var Compose = (function() {
     lastScrollPosition: 0,
     resizing: false,
 
-    // Stop further input because the max size is exceeded
-    locked: false,
-
     // 'sms' or 'mms'
     type: 'sms',
 
+    lock: {
+      send: false,
+      edit: false,
+      type: null
+    },
```

Comment:
I'd prefer to have functions used to retrieve the values, than properties that an external object needs to set (even through a method).